src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLogger.cs

```diff
@@ -41,19 +45,21 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
         {
             if (this.IsEnabled(logLevel))
             {
-                var dict = new Dictionary<string, string>();
+                TraceTelemetry traceTelemetry = new TraceTelemetry(formatter(state, exception), this.GetSeverityLevel(logLevel));
+                IDictionary<string, string> dict = traceTelemetry.Context.Properties;
                 dict["CategoryName"] = this.categoryName;
                 dict["Exception"] = exception?.ToString();
-                var stateDictionary = state as IReadOnlyList<KeyValuePair<string, object>>;
+                IReadOnlyList<KeyValuePair<string, object>> stateDictionary = state as IReadOnlyList<KeyValuePair<string, object>>;
                 if (stateDictionary != null)
                 {
-                    foreach (var item in stateDictionary)
+                    foreach (KeyValuePair<string, object> item in stateDictionary)
                     {
                         dict[item.Key] = Convert.ToString(item.Value);
                     }
                 }
 
-                this.telemetryClient.TrackTrace(formatter(state, exception), this.GetSeverityLevel(logLevel), dict);
+                traceTelemetry.Context.GetInternalContext().SdkVersion = this.sdkVersion;
```

Comment:
Initializers will set the version for all telemetry item - custom reported, collected by dependency collector module, etc. We want to distinguish those so we can help troubleshoot the issue when we see "bad" trace telemetry. Knowing how that "bad" trace was collected is essential